terraform/modules/common/ami/main.tf

```diff
@@ -0,0 +1,33 @@
+## Variables
+
+variable "amazon_release" {
+  description = "The amazon release used to generate the encrypted ami"
+}
+
+variable "stack_name" {
+  description = "The name of the stack the ami will belong to"
+}
+
+## Data sources
+
+data "aws_ami" "source" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["amzn-ami-${var.amazon_release}-amazon-ecs-optimized"]
```

Comment:
It feels like this ami module is specific to ECS optimised instances so maybe the module should be called `ecs-ami` rather than just `ami`?

Reply:
I'll have a chat with Ed to see what he thinks